contributors/design-proposals/storage/csi-migration.md

```diff
@@ -287,8 +287,29 @@ existing Pods in the ADC.
 
 
 ### Volume Resize
-
-TODO: Design
+#### Offline Resizing
+In the OperationGenerator, `GenerateExpandVolumeFunc` is used to expand a volume after 
+Pod is deleted. At the beginning of the `GenerateExpandVolumeFunc`, we will check whether 
+the volume supports migration and whether migration is enabled. If enabled, we will return
+a noop function to by-pass in-tree resizer (expand controller).
+
+In the OperationGenerator, `GenerateMountVolumeFunc` is used to expand file system after volume
+is expanded and staged/mounted. The migration logic is covered by previous migration of volume mount.
+
+For external resizer, we will update it to expand volume for both CSI volume and in-tree 
+volume (only if migration is enabled). For migrated in-tree volume, it will update in-tree PV object
+with new volume size and mark in-tree PVC as resizing finished.
+
+To synchronize between in-tree resizer and external resize, external resizer will find resizer name
+using PVC annotation `volume.kubernetes.io/storage-resizer`. Since `volume.kubernetes.io/storage-resizer`
+annotation defines the stroage backend actually being used, for in-tree volume, the resizer name is the
```

Comment:
what about when we migrate the in-tree volume. Will this be the translated CSI driver name? Does the External-Resizer "know" or "need to know" whether this is a native CSI volume or a migrated in-tree volume?

Reply:
yeah the name should be translated CSI driver name, so as exernal-resizer can know if it should resize this PVC.